Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/FilterAdapter.cpp

```diff
@@ -0,0 +1,501 @@
+            }
+            auto addedChild = parentMatchState->m_childMatchState[newRowIndex];
+
+            // recursively add descendents and cache their match status
+            auto sourceContents = m_sourceAdapter->GetContents();
+            recursivelyRegenerateMatches(addedChild, sourceContents[sourcePath]);
+        }
+    }
+
+    void RowFilterAdapter::UpdateMatchState(MatchInfoNode* rowState)
+    {
+        auto includeInFilter = [](MatchInfoNode* node) -> bool
+        {
+            return (node->m_matchesSelf || !node->m_matchingDescendents.empty());
+        };
+        bool usedToMatch = includeInFilter(rowState);
+
+        rowState->m_matchesSelf = MatchesFilter(rowState); //update own match state
+
+        auto& matchingChildren = rowState->m_matchingDescendents;
+        for (auto childIter = matchingChildren.begin(); childIter != matchingChildren.end(); /* omitted */)
+        {
+            // it's a child's job to update its parents, but a parent still has to check if any
+            // formerly matching children have been removed
+            auto& currChildren = rowState->m_childMatchState;
+            if (AZStd::find(currChildren.begin(), currChildren.end(), *childIter) == currChildren.end())
+            {
+                childIter = matchingChildren.erase(childIter);
+            }
+            else
+            {
+                ++childIter;
+            }
+
+        }
+
+        // update each ancestor until updating their m_matchingDescendents doesn't change their inclusion state
+        MatchInfoNode* currState = rowState;
+        bool nowMatches = includeInFilter(currState);
+        while (currState->m_parentNode && nowMatches != usedToMatch)
+        {
+            usedToMatch = includeInFilter(currState->m_parentNode);
+            if (nowMatches)
+            {
+                currState->m_parentNode->m_matchingDescendents.insert(currState);
+            }
+            else
+            {
+                currState->m_parentNode->m_matchingDescendents.erase(currState);
+            }
+            nowMatches = includeInFilter(currState->m_parentNode);
+            currState = currState->m_parentNode;
+        }
+    }
+
+    Dom::Path RowFilterAdapter::GetRowPath(const Dom::Path& sourcePath) const
+    {
+        Dom::Path rowPath;
+        auto currDomValue = m_sourceAdapter->GetContents();
+        for (const auto& pathEntry : sourcePath)
+        {
+            currDomValue = currDomValue[pathEntry];
+            if (IsRow(currDomValue))
+            {
+                rowPath.Push(pathEntry);
+            }
+        }
+        return rowPath;
+    }
+
+    ValueStringFilter::ValueStringFilter()
+        : RowFilterAdapter()
+    {
+    }
+
+    void ValueStringFilter::SetFilterString(QString filterString)
+    {
+        if (m_filterString != filterString)
+        {
+            m_filterString = std::move(filterString);
```

Comment:
nit: `AZStd::move` (a linter for AZStd/std violations might be kind of cool?)

Reply:
fixed, thanks!